apps/app/src/app/pages/pages/dashboard/dashboard.component.spec.ts

```diff
@@ -52,54 +116,136 @@ describe('DashboardComponent', () => {
+    });
   });
 
-  it('should contains "app-pages-dashboard-application-card" in a tag with class "clr-row" after first h2', () => {
-    component.applications = [
-      {
-        _id: {
-          applicationName: 'application_1',
-          applicationVersion: '1.0.0',
-        },
-      },
-    ];
-    fixture.detectChanges();
+  describe('applications', () => {
+    it('should get all applications with errors', () => {
+      component.onRefreshApplicationsErrors({});
 
-    const compiled = fixture.debugElement.nativeElement;
-    expect(compiled.querySelector('div.clr-row').textContent).toContain(
-      'application_1'
-    );
+      expect(component.applicationsErrors).toEqual(applicationsErrorsPaginated);
+    });
+
+    it('should have an error when getting applications with errors failed', () => {
+      const error = { status: 404, operation: 'getAllWithErrorsPaginated' };
+
+      http.getAllWithErrorsPaginated.and.returnValue(
+        throwError(() => {
+          return error;
+        })
+      );
+
+      component.onRefreshApplicationsErrors({});
+
+      expect(component.errors[0]).toEqual(error);
+    });
+
+    it('should add clicked application to currentApplication and navigate', () => {
+      const application = applications[0];
+
+      const settingsService = TestBed.inject(SettingsService);
+      const router = TestBed.inject(Router);
+
+      const spySettings = spyOn(
+        settingsService,
+        'addCurrentApplication'
+      ).and.returnValues();
+      const spyRouter = spyOn(router, 'navigate').and.returnValues(
+        new Promise((resolve) => resolve(true))
+      );
+
+      component.onApplicationClick(application);
+
+      expect(spySettings).toHaveBeenCalledWith(application);
+      expect(spyRouter).toHaveBeenCalledWith([
+        '/',
+        'applications',
+        application._id.applicationName,
+        application._id.applicationVersion,
+        'sessions',
+      ]);
+    });
   });
 
-  it('should contains a app-alert-error after the second h2', () => {
-    const compiled = fixture.debugElement.nativeElement;
-    expect(compiled.querySelector('h2 + app-alert-error')).toBeTruthy();
+  describe('Seq', () => {
+    it('should open a _blank tab with seq url', () => {
+      const taskId = '1';
+      const url = 'test-url';
+      const spy = spyOn(window, 'open');
+      const settingsService = TestBed.inject(SettingsService);
+      spyOn(settingsService, 'generateSeqUrlForTaskError').and.returnValue(url);
+
+      component.redirectToSeq(taskId);
+
+      expect(spy).toHaveBeenCalledWith(url, '_blank');
+    });
   });
 
-  it('should contains a app-pages-dashboard-applications-errors-list after the app-alert-error', () => {
-    const compiled = fixture.debugElement.nativeElement;
-    expect(
-      compiled.querySelector(
-        'app-alert-error + app-pages-dashboard-applications-errors-list'
-      )
-    ).toBeTruthy();
+  describe('ui', () => {
```

Comment:
Pareil ici je suis pas sûre que ce soit utile de tester la présence de ces composants ni leur contenu mais plutôt de tester :
- (applicationChange)="onApplicationClick($event)"
- (clickSeqLink)="redirectToSeq($event)"
- (refresh)="onRefreshApplicationsErrors($event)"
Si ce n'est pas possible de tester ces output alors ne fais aucun test dans ui :)

Reply:
Okkk, je vois !

Reply:
Du coup, tout ça, c'est bien testé !

Reply:
https://stackoverflow.com/questions/56641850/how-to-test-child-component-output